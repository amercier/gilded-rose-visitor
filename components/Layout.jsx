import React from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import styled from 'styled-components';
import FadeIn from 'react-fade-in';
import ConnectedHeader from './Header';

const Container = styled.div`
  width: 100%;
  max-width: 64rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 2rem;
`;

/**
 * App layout.
 *
 * @param {*} children - Child components.
 * @returns {React.Element} The rendered element.
 */
const Layout = ({ children }) => (
  <>
    <CssBaseline />
    <ConnectedHeader />
    <Container>
      <FadeIn>{children}</FadeIn>
    </Container>
  </>
);

export default Layout;
