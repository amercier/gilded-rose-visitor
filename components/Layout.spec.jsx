import React from 'react';
import { shallow } from 'enzyme';
import Layout from './Layout';

describe('Layout', () => {
  it('renders a list of items', () => {
    const index = shallow(<Layout>CONTENT</Layout>);
    expect(index).toMatchSnapshot();
  });
});
