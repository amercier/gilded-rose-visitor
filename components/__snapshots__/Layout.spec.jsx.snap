// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders a list of items 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <Connect(Header) />
  <Layout__Container>
    CONTENT
  </Layout__Container>
</Fragment>
`;
